server/src/service.rs

```diff
@@ -112,20 +112,23 @@ impl Service {
         Ok((unlocked, locked))
     }
 
-    #[zbus(out_args("unlocked", "prompt"))]
     pub async fn unlock(
         &mut self,
-        _objects: Vec<OwnedObjectPath>,
-    ) -> Result<(Vec<OwnedObjectPath>, ObjectPath), ServiceError> {
-        todo!()
+        objects: Vec<OwnedObjectPath>,
+    ) -> Result<(Vec<OwnedObjectPath>, OwnedObjectPath), ServiceError> {
+        let (unlocked, _not_unlocked) = self.set_locked(false, &objects).await?;
+
```

Comment:
I would have still called the methods from Collection and Item so you don't need the unused escape. Things can be changed later on as I mentioned previously

Reply:
Done.